src/test/java/com/conveyal/datatools/manager/jobs/AutoDeployFeedJobTest.java

```diff
@@ -0,0 +1,206 @@
+        user = Auth0UserProfile.createTestAdminUser();
+
+        server = createOtpServer();
+        DeployJob.DeploySummary deploySummary = createDeploymentSummary(server.id);
+        project = createProject();
+        deployment = createDeployment(deploySummary, project.id);
+        mockFeedSource = createFeedSource("Mock Feed Source", project.id);
+
+        feedVersion = createFeedVersion(mockFeedSource);
+        feedVersionExpireIn2099 = TestUtils.createFeedVersion(mockFeedSource,
+            TestUtils.zipFolderFiles("fake-agency-expire-in-2099"));
+        feedVersionExpireIn2099UnusedRoute = TestUtils.createFeedVersion(mockFeedSource,
+            TestUtils.zipFolderFiles("fake-agency-with-only-calendar-expire-in-2099-with-unused-route"));
+
+        fetchStillInProgress_server = createOtpServer();
+        DeployJob.DeploySummary deploySummaryFetchInProgress = createDeploymentSummary(fetchStillInProgress_server.id);
+        fetchStillInProgress_project = createProject();
+        fetchStillInProgress_deployment = createDeployment(deploySummaryFetchInProgress, fetchStillInProgress_project.id);
+        fetchStillInProgress_mockFeedSourceFakeJob =
+            createFeedSource("Mock Feed Source For Fetch In Progress Fake Job", fetchStillInProgress_project.id);
+        fetchStillInProgress_mockFeedSource =
+            createFeedSource("Mock Feed Source For Fetch In Progress", fetchStillInProgress_project.id);
+        fetchStillInProgress_feedVersion = getFeedVersionFromGTFSFile(fetchStillInProgress_mockFeedSourceFakeJob,
+            TestUtils.zipFolderFiles("fake-agency-expire-in-2099"));
+        Persistence.feedVersions.create(fetchStillInProgress_feedVersion);
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        Auth0Connection.setAuthDisabled(Auth0Connection.getDefaultAuthDisabled());
+        server.delete();
+        deployment.delete();
+        mockFeedSource.delete();
+        feedVersionExpireIn2099.delete();
+        feedVersionExpireIn2099UnusedRoute.delete();
+
+        fetchStillInProgress_mockFeedSource.delete();
+        fetchStillInProgress_server.delete();
+        fetchStillInProgress_deployment.delete();
+        fetchStillInProgress_mockFeedSourceFakeJob.delete();
+        fetchStillInProgress_feedVersion.delete();
+
+        Persistence.projects.removeById(fetchStillInProgress_project.id);
+        Persistence.projects.removeById(project.id);
+        Persistence.feedVersions.removeById(feedVersion.id);
+        Persistence.feedVersions.removeById(fetchStillInProgress_feedVersion.id);
+    }
+
+    // TODO: AutoDeployFeedJob now processes multiple projects. It is not possible to set the job status in
+    //  relation to a single project success or failure. This means asserting on a change of state to the class project
+    //  e.g. lastAutoDeploy (as below). But this doesn't seem right and doesn't have the required granularity.
+    //  A few options come to mind:
```

Comment:
Making an assertion on the autoDeployJob.status.completed or autoDeployJob.status.message could suffice.

Reply:
Actually, I see what you're saying now since the job should always complete successfully. I think checking the lastAutoDeploy is fine for now and that creating extra fields for the purposes of testing wouldn't add much value.